core/go/pkg/persistence/gorm.go

```diff
@@ -217,3 +227,142 @@ func (gp *provider) Transaction(parentCtx context.Context, fn func(ctx context.C
 func (gp *provider) NOTX() DBTX {
 	return newNOTX(gp.gdb)
 }
+
+var (
+	whereClause     = clause.Where{}.Name()
+	postgresDialect = postgres.Dialector{}.Name()
+)
+
+// ANY is a custom implementation of the clause.IN or clause.Expr which binds array of Values directly to a single variable
+// In the clause.Expr, it replaces the IN clause with ANY, which is a PostgreSQL-specific feature.
+// it has been implemented to address the limitation of "protocol limited to 65535 parameters".
+type ANY struct {
+	IN   *clause.IN
+	Expr *clause.Expr
+}
+
+// UseAny configures the DB to use the ANY type for IN clauses to resolve parameter limitations.
+func UseAny(db *gorm.DB) {
+	currentDialect := db.Dialector.Name()
+	if currentDialect != postgresDialect {
+		log.L(db.Statement.Context).Errorf("ANY clause not supported with %q dialect", currentDialect)
+		return
+	}
+
+	db.ClauseBuilders[whereClause] = func(c clause.Clause, builder clause.Builder) {
+		where := c.Expression.(clause.Where)
+		for i, expr := range where.Exprs {
+			if in, ok := expr.(clause.IN); ok {
+				where.Exprs[i] = ANY{IN: &in}
+			}
+
+			if expr, ok := expr.(clause.Expr); ok {
```

Comment:
Hit another issue where this doesn't handle nested expr so things like 	

`"SELECT count(*) FROM \"test\" WHERE tag = 'a' AND masked = 1 AND sequence != 999 AND correl_id IS NULL AND sequence > 10 AND ((masked = 1 AND tag IN ('a','b','c') AND tag NOT IN ('x','y') AND tag NOT IN ('z')) OR masked = 0) LIMIT 10", generatedSQL)`